src/com/company/Main.java

```diff
@@ -20,53 +21,55 @@ public static void main(String[] args) {
 
         isFileFormatSupported(file);
 
-        if(!file.exists() || !isFileFormatSupported(file)) {
+        if (!file.exists() || !isFileFormatSupported(file)) {
             System.out.println("file is not valid");
-            return ;
+            return;
         }
-        if(file.length() == 0) {
+        if (file.length() == 0) {
             System.out.println("the file does not contains any data ");
-            return ;
+            return;
         }
         System.out.println("processing................");
         String data = readFileAsString(txtFilePath);
         searchTheWord(data, searchword);
     }
-        public static String readFileAsString(String fileName) {
-                String data = "";
-              try {
-                   data = new String(Files.readAllBytes(Paths.get(fileName)));
-              } catch (IOException e) {
-                  e.printStackTrace();
-              }
-              return data.replaceAll("[^a-zA-Z0-9@]", " ");
-    }
-        public static boolean isFileFormatSupported(File file) {
-           return (file.getName().endsWith(".txt"));
+
+    public static String readFileAsString(String fileName) {
+        String data = "";
+        try {
+            data = new String(Files.readAllBytes(Paths.get(fileName)));
+        } catch (IOException e) {
+            e.printStackTrace();
         }
-        public static void searchTheWord(String data,String searchword) {
-                StringTokenizer st = new StringTokenizer(data);
-                int count = 0;
+        return data.replaceAll("[^a-zA-Z0-9@]", " ");
```

Comment:
Instead of hardcoding regex here, you could probably make this is a final variable?

Reply:
created the variable and made that as you suggested.